README_V8.md

```diff
@@ -0,0 +1,549 @@
+
+### Retry
+
+<!-- snippet: retry -->
+```cs
+// To use the default retry options with pre-configured settings:
+//
+// MaxRetryAttempts: 3,
+// Delay: 2 seconds,
+// Backoff Type: Constant,
+// UseJitter: false,
+// ShouldHandle: Handles all exceptions except OperationCanceledException
+new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions());
+
+// For instant retries with no delay
+new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
+{
+    Delay = TimeSpan.Zero
+});
+
+// For advanced control over the retry behavior, including the number of attempts,
+// delay between retries, and the types of exceptions to handle.
+new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
+{
+    ShouldHandle = new PredicateBuilder().Handle<SomeExceptionType>(),
+    BackoffType = DelayBackoffType.Exponential,
+    UseJitter = true,  // Adds a random factor to the delay
+    MaxRetryAttempts = 4,
+    Delay = TimeSpan.FromSeconds(3),
+});
+
+// To use a custom function to generate the delay for retries
+new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
+{
+    MaxRetryAttempts = 2,
+    DelayGenerator = args =>
+    {
+        var delay = args.AttemptNumber switch
+        {
+            0 => TimeSpan.Zero,
+            1 => TimeSpan.FromSeconds(1),
+            _ => TimeSpan.FromSeconds(5)
+        };
+
+        // This example uses a synchronous delay generator,
+        // but the API also supports asynchronous implementations.
+        return new ValueTask<TimeSpan?>(delay);
+    }
+});
+
+// To extract the delay from the result object
+new ResiliencePipelineBuilder<HttpResponseMessage>().AddRetry(new RetryStrategyOptions<HttpResponseMessage>
+{
+    DelayGenerator = args =>
+    {
+        if (args.Outcome.Result is HttpResponseMessage responseMessage &&
+            TryGetDelay(responseMessage, out TimeSpan delay))
+        {
+            return new ValueTask<TimeSpan?>(delay);
+        }
+
+        // Returning null means the retry strategy will use its internal delay for this attempt.
+        return new ValueTask<TimeSpan?>((TimeSpan?)null);
+    }
+});
+
+// To get notifications when a retry is performed
+new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
+{
+    MaxRetryAttempts = 2,
+    OnRetry = args =>
+    {
+        Console.WriteLine("OnRetry, Attempt: {0}", args.AttemptNumber);
+
+        // Event handlers can be asynchronous; here, we return an empty ValueTask.
+        return default;
+    }
+});
+
+// To keep retrying indefinitely until successful
+new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
+{
+    MaxRetryAttempts = int.MaxValue,
+});
+```
+<!-- endSnippet -->
+
+If all retries fail, a retry strategy rethrows the final exception back to the calling code.  For more depth visit [retry strategy documentation](https://github.com/App-vNext/Polly/wiki/Retry).
```

Comment:
Should this point to a newer stub too? Similar comments throughout this file to avoid linking to the Wiki in these new files.

Reply:
Will update these in follow-up